packages/appsync-modelgen-plugin/src/visitors/appsync-dart-visitor.ts

```diff
@@ -510,6 +510,10 @@ export class AppSyncModelDartVisitor<
               return field.isList
                 ? `${fieldName} = (json['${varName}'] as List<dynamic>)?.map((dynamic e) => e is double ? e.toInt() : e as int)?.toList()`
                 : `${fieldName} = json['${varName}']`;
+            case this.scalars['Float']:
+              return field.isList
+                ? `${fieldName} = json['${varName}']?.cast<${this.getNativeType({ ...field, isList: false })}>()`
+                : `${fieldName} = (json['${varName}'] as num?)?.toDouble()`;
```

Comment:
The null check pattern looks inconsistent across the cases. For example, in case of Timestamps, it is `json['${varName}'] != null ? ${fieldNativeType}.fromSeconds(json['${varName}']) : null`

Reply:
why is this explicit casting to `num` and then to `Double` necessary?

Reply:
This line was taken from the comment of the original issue. Need to confirm with Flutter team

Reply:
Hmm, rather than casting the list type (which is the default case last resort), I think we should be able to keep the same pattern as most of these other scalars. Since I noticed customers also having trouble with int types, I suggest improving the `Int` case and basically duplicating it for `Float`:
```dart
case this.scalars['Int']:
  return field.isList
    ? `${fieldName} = (json['${varName}'] as List?)?.map((e) => (e as num).toInt()).toList()`
    : `${fieldName} = (json['${varName}'] as num?)?.toInt()`;
case this.scalars['Float']:
  return field.isList
    ? `${fieldName} = (json['${varName}'] as List?)?.map((e) => (e as num).toDouble()).toList()`
    : `${fieldName} = (json['${varName}'] as num?)?.toDouble()`;
```
Technically, we may not need to cast to `num` first since it the compiler would just treat it as dynamic but the cast helps it understand that `toInt()` and `toDouble()` are indeed member functions